profiles/models.py

```diff
@@ -99,6 +100,25 @@ class Profile(models.Model):
     mailing_address = models.TextField(blank=True, null=True)
     date_joined_micromasters = models.DateTimeField(blank=True, null=True, auto_now_add=True)
     linkedin = JSONField(blank=True, null=True)
+    student_id = models.IntegerField(blank=True, null=True, unique=True)
+
+    @transaction.atomic
+    def save(self, *args, **kwargs):
+        """need to handle setting the student_id number"""
+        if self.id is None or self.student_id is None:
+            max_id = Profile.objects.aggregate(Max('student_id'))
+            if max_id['student_id__max'] is None:
+                max_id = 1
+            else:
+                max_id = max_id['student_id__max']
+            while Profile.objects.filter(student_id=max_id).exists():
+                max_id += 1
+            self.student_id = max_id
+        super(Profile, self).save(*args, **kwargs)
 
     def __str__(self):
         return 'Profile for "{0}"'.format(self.user.username)
+
+    def pretty_printed_student_id(self):
+        """pretty prints the student id for easy display"""
+        return "MMM{0:06}".format(self.student_id)
```

Comment:
I would make this a property method:

```
@property
def pretty_printed_student_id(self):
```

so you can access it as a property and not a method:
`user_profile. pretty_printed_student_id`